src/palimpzest/corelib/functions.py

```diff
@@ -38,6 +38,7 @@ def html_to_text_with_links(self, html):
     def map(self, dr: DataRecord):
         textcontent = requests.get(dr.url).text
         dr2 = DataRecord(self.outputSchema)
```

Comment:
Seems kind of error-prone to have a DR constructor that requires we always follow it up with a parent assignment.  Shouldn't parent assignment be required by default?

Reply:
Good catch, I added a `parent_uuid` kwarg and didn't update it here, will push that in a second.